packages/firestore/e2e/withConverter.e2e.js

```diff
@@ -0,0 +1,615 @@
+/*
+ * Copyright (c) 2021-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+const COLLECTION = 'firestore';
+const { wipe } = require('./helpers');
+
+const {
+  getFirestore,
+  doc,
+  collection,
+  refEqual,
+  addDoc,
+  setDoc,
+  getDoc,
+  query,
+  where,
+  getDocs,
+  writeBatch,
+  increment,
+  initializeFirestore,
+} = firestoreModular;
+
+// Used for testing the FirestoreDataConverter.
+class Post {
+  constructor(title, author, id = 1) {
+    this.title = title;
+    this.author = author;
+    this.id = id;
+  }
+  byline() {
+    return this.title + ', by ' + this.author;
+  }
+}
+
+const postConverter = {
+  toFirestore(post) {
+    return { title: post.title, author: post.author };
+  },
+  fromFirestore(snapshot) {
+    const data = snapshot.data();
+    return new Post(data.title, data.author);
+  },
+};
+
+const postConverterMerge = {
+  toFirestore(post, options) {
+    if (
+      options &&
+      ((options && options.merge === true) ||
+        (options && Array.isArray(options.mergeFields) && options.mergeFields.length > 0))
+    ) {
+      post.should.not.be.an.instanceof(Post);
+    } else {
+      post.should.be.an.instanceof(Post);
+    }
+    const result = {};
+    if (post.title) {
+      result.title = post.title;
+    }
+    if (post.author) {
+      result.author = post.author;
+    }
+    return result;
+  },
+  fromFirestore(snapshot) {
+    const data = snapshot.data();
+    return new Post(data.title, data.author);
+  },
+};
+
+// v8 compatibility helper functions
+function modifyIgnoreUndefinedProperties(db, value) {
+  // JS SDK settings can only be called once
+  if (Platform.other) {
+    db._settings.ignoreUndefinedProperties = value;
+  } else {
+    db.settings({ ignoreUndefinedProperties: value });
+  }
```

Comment:
Still can't believe that was necessary - such a trivial part of this whole work and ended up being a pain. Sorry